apps/framr-web/lib/services/core/modules/tools/ToolsService.ts

```diff
@@ -2,41 +2,37 @@ import { CreateTool, MWDTool } from '../../../../types';
 import { ToolEnum } from '../../../../types/enums';
 import { FramrServiceError } from '../../../libs/errors';
 import { EventBus, EventBusChannelStatus } from '../../../libs/event-bus';
-import { IDBFactory, StoreRecord } from '../../../libs/idb';
+import { IDBFactory } from '../../../libs/idb';
 import { IDBConnection } from '../../db/IDBConnection';
-import { FramrDBSchema } from '../../db/schema';
+import { FramrDBSchema, ToolRecord } from '../../db/schema';
 
 import { ToolInterface, ToolsEventChannel } from './ToolInterface';
 
 export class ToolsService implements ToolInterface {
   private readonly eventBus: EventBus;
   private readonly database: IDBFactory<FramrDBSchema>;
+  private readonly STORE_NAME = 'tools';
 
   constructor() {
     this.database = IDBConnection.getDatabase();
     this.eventBus = new EventBus();
   }
 
   create(createTool: CreateTool): void {
-    let newTool: StoreRecord<FramrDBSchema, 'tools'> | null = null;
+    let newTool: ToolRecord | null = null;
     let channel = ToolsEventChannel.CREATE_TOOLS_CHANNEL;
 
-    if (createTool.type == ToolEnum.LWD) {
-      newTool = {
-        value: { ...createTool, id: crypto.randomUUID(), type: ToolEnum.LWD },
-      };
-    } else {
-      newTool = {
-        value: {
-          ...(createTool as MWDTool),
-          id: crypto.randomUUID(),
-          type: ToolEnum.MWD,
-        },
-      };
-    }
+    newTool = {
```

Comment:
Was it not obvious ?
```suggestion
   let newTool: ToolRecord = {
```